platform/security/handler/security-handler-saml/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -422,33 +424,29 @@ public Response postLogoutRequest(
         LOGGER.info(UNABLE_TO_VALIDATE_LOGOUT_REQUEST, e);
         return buildLogoutResponse(UNABLE_TO_VALIDATE_LOGOUT_REQUEST);
       }
-    } else {
-      try {
-        LogoutWrapper<LogoutResponse> logoutResponse =
-            logoutMessage.extractSamlLogoutResponse(samlSecurity.base64Decode(encodedSamlResponse));
-        if (logoutResponse == null) {
-          LOGGER.info(UNABLE_TO_PARSE_LOGOUT_RESPONSE);
-          return buildLogoutResponse(UNABLE_TO_PARSE_LOGOUT_RESPONSE);
-        }
-        new SamlValidator.Builder(simpleSign)
-            .buildAndValidate(
-                request.getRequestURL().toString(),
-                SamlProtocol.Binding.HTTP_POST,
-                logoutResponse.getMessage());
-      } catch (ValidationException e) {
-        LOGGER.info(UNABLE_TO_VALIDATE_LOGOUT_RESPONSE, e);
-        return buildLogoutResponse(UNABLE_TO_VALIDATE_LOGOUT_RESPONSE);
-      } catch (LogoutSecurityException | XMLStreamException e) {
-        LOGGER.info(UNABLE_TO_PARSE_LOGOUT_RESPONSE, e);
+    }
+
+    try {
+      LogoutWrapper<LogoutResponse> logoutResponse =
+          logoutMessage.extractSamlLogoutResponse(samlSecurity.base64Decode(encodedSamlResponse));
```

Comment:
I know we didn't do it before but we should verify that `encodedSamlRequest != null` here

Reply:
Just pushed a commit, but I'm not sure I returned the right response. Does it look right to you?
```java
if (encodedSamlResponse == null) {
  return Response.serverError().entity(SYSTEM_CANNOT_DECODE_REQUEST).build();
}
```

Reply:
I updated the message
```java
if (encodedSamlResponse == null) {
  return Response.serverError()
      .entity("System did not receive a SAMLRequest or a SAMLResponse to process")
      .build();
}
```